### PR TITLE
feat: [SRM-12423]: Onboarding API changes to use AWS Prometheus.

### PR DESCRIPTION
**Pull request by dhruv-harness**
## Describe your changes
Extending the existing PrometheusServiceImpl.java class functionality to support Aws Prometheus as health-source. Using the DataSourceType parameter to determine whether to use Aws connection vs direct connection. If using Aws, validating the required input fields.
I have moved the data parsing logic for meta-data and sample metric data from DSL scripts to the service layer class. This brings reusability to the parsing code and keeps the DSL short. 
NOTE: We can reduce the number of Prometheus DSLs to 1. Currently we have 4.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/39090)

### CODE REVIEW

File: 300-cv-nextgen/src/main/java/io/harness/cvng/core/beans/params/PrometheusConnectionParams.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import lombok.experimental.FieldDefaults;
+import lombok.experimental.SuperBuilder;
+
+@FieldDefaults(level = AccessLevel.PRIVATE)
+@Data
+@SuperBuilder
+@NoArgsConstructor
+@AllArgsConstructor
+public class PrometheusConnectionParams {
+  @QueryParam("region") String region;
```
Comment (n00bitax) [inline 300-cv-nextgen/src/main/java/io/harness/cvng/core/beans/params/PrometheusConnectionParams.java:29]: Is this an AWS Prometheus-specific class or a generic one? .... we should call it is AWSRegion if its only applicable for AWS.

Comment (dhruv-harness) [reply 300-cv-nextgen/src/main/java/io/harness/cvng/core/beans/params/PrometheusConnectionParams.java:29]: Specific to Prometheus use-cases but generic for any connection params.

File: 300-cv-nextgen/src/main/java/io/harness/cvng/core/resources/PrometheusResource.java [hunk 2]
```diff
@@ -51,10 +52,10 @@ public class PrometheusResource {
   @ExceptionMetered
   @ApiOperation(value = "get all metric names", nickname = "getMetricNames")
```
Comment (n00bitax) [inline 300-cv-nextgen/src/main/java/io/harness/cvng/core/resources/PrometheusResource.java:53]: Where are we getting the metric names from? Can we specify if it's from our own DB or through AWS API or whatever it is.

File: 300-cv-nextgen/src/main/java/io/harness/cvng/core/beans/params/PrometheusConnectionParams.java (hunk 1 above)
Comment (n00bitax) [reply 300-cv-nextgen/src/main/java/io/harness/cvng/core/beans/params/PrometheusConnectionParams.java:29]: okay then we shouldnt change

**Review by n00bitax — APPROVED** [review]

File: 300-cv-nextgen/src/main/java/io/harness/cvng/core/services/impl/PrometheusServiceImpl.java [hunk 3]
```diff
@@ -72,34 +178,32 @@ private List<String> callOnboardingServiceForStringList(DataCollectionRequest re
...
+  }
 
-    OnboardingResponseDTO response = onboardingService.getOnboardingResponse(accountId, onboardingRequestDTO);
-    final Gson gson = new Gson();
-    Type type = new TypeToken<List<PrometheusSampleData>>() {}.getType();
-    return gson.fromJson(JsonUtils.asJson(response.getResult()), type);
+  private List<PrometheusSampleData> toListOfPrometheusSampleData(OnboardingResponseDTO onboardingResponseDTO) {
+    List<PrometheusSampleData> sampleData = new ArrayList<>();
+    List<Map<String, Object>> responseData =
+        (List<Map<String, Object>>) ((Map<String, Object>) (((Map<String, Object>) onboardingResponseDTO.getResult())
```
Comment (n00bitax) [inline 300-cv-nextgen/src/main/java/io/harness/cvng/core/services/impl/PrometheusServiceImpl.java:191]: The response might not always be what we want, if it's a 500 error or connection timed out ..... we should make sure we have retries for the call, check the status code to be 200 OK and the repsonse is parsable within a try-catch block.

Comment (dhruv-harness) [reply 300-cv-nextgen/src/main/java/io/harness/cvng/core/services/impl/PrometheusServiceImpl.java]: The http request executor takes care of the retries. Added try catch block around the parsing.

**Review by AbhijithHarness — APPROVED** [review]
> Reviewed 8 of 13 files at r1, 4 of 5 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39090)* status: 3. unresolved discussions. 

___
*[`300-cv-nextgen/src/main/java/io/harness/cvng/core/beans/params/PrometheusConnectionParams.java` line 29 at r2](https://reviewable.io/reviews/harness/harness-core/39090#-NEttpMKRIyeJKI0c0XM-r2-29:-NFDIkex48Tk1HBV2AWM:b3gt6dv) ([raw file](https://github.com/harness/harness-core/blob/87d9db1e2497bee141582049cc554a0962400eaf/300-cv-nextgen/src/main/java/io/harness/cvng/core/beans/params/PrometheusConnectionParams.java#L29)):*
<details><summary><i>Previously, n00bitax (Ansuman Satapathy) wrote…</i></summary><blockquote>

okay then we shouldnt change
</blockquote></details>

nit: add some

**Review by n00bitax — APPROVED** [review]

**Pull request merged**